miden/src/cli/run.rs

```diff
@@ -31,13 +32,14 @@ impl RunCmd {
         let script = ScriptFile::read(&self.assembly_file)?;
 
         // load input data from file
-        let input_data = InputFile::read(&self.input_file, &self.assembly_file)?;
+        let input_data = InputFile::read(&self.input_file, &self.assembly_file)
+            .map_or(ProgramInputs::none(), |input| input.get_program_inputs());
```

Comment:
I'm not sure this will work. I believe this will set inputs to `ProgramInputs::none()` in case of any error. What we want is to get `ProgramInputs::none()` when the input file wasn't specified and doesn't exist. In all other cases, this should still return an error.

My initial thinking is that we may need to move away from using `InputFile` struct directly. Maybe it is better to have a function - e.g.:
```Rust
pub fn read_inputs(inputs_path: &Option<PathBuf>, script_path: &Path) -> Result<PorgramInputs, String>
```

Reply:
Thanks for catching that mistake.
I've added this check in the same `read` function in `InputFile` struct where if a file is not specified and is not present, it sets `stack_init` to an empty vector.